keras_cv/layers/preprocessing/channel_shuffle.py

```diff
@@ -52,15 +51,10 @@ def __init__(self, groups=3, seed=None, **kwargs):
         self.groups = groups
         self.seed = seed
 
-    def _channel_shuffling(self, images):
-        unbatched = images.shape.rank == 3
-
-        if unbatched:
-            images = tf.expand_dims(images, axis=0)
-
-        height = tf.shape(images)[1]
-        width = tf.shape(images)[2]
-        num_channels = images.shape[3]
+    def augment_image(self, image, transformation=None):
+        height = tf.shape(image)[0]
+        width = tf.shape(image)[1]
+        num_channels = image.shape[2]
 
         if not num_channels % self.groups == 0:
```

Comment:
If you want to do a runtime check you must run `tf.debugging.Assert` instead of using an if statement.  This will fix the above issue too.

Reply:
Although, this should be a solveable issue at trace time - I see.

Reply:
Exactly, when I use `tf.debugging.Assert` and  `tf.debugging.assert_equal`, the assertions get bypassed  at trace time and the function raises an error during reshaping.

When I use `tf.cond`  and `if`, the raise function just gets executed as part of tracing thus throwing an exception even when the code is completely fine.

Reply:
@LukeWood Would love any inputs on how to fix this!^